File: InnerBeauty/app/src/main/java/com/boostcamp/eunjilee/innerbeauty/model/PlayModel.java
package com.boostcamp.eunjilee.innerbeauty.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Created by eunjilee on 06/02/2017.
 */

public class PlayModel implements Serializable {
    @SerializedName("play_id")
    private int playId;
    @SerializedName("play_code")
    private int playCode;
    @SerializedName("play_title")
    private String playTitle;
    @SerializedName("play_actors") //TODO : Actor객체로 바꿔야하지 않을까. Actor 테이블을 만들어서 join해서 사용해야하나?
    private String playActors;
    @SerializedName("start_date")
    private  String startDate;
    @SerializedName("end_date")
    private  String endDate;
    @SerializedName("close_date")
    private  String closeDate;
    @SerializedName("play_time")
    private  String playTime;
    @SerializedName("play_run_time")
    private  String playRunTime;
    @SerializedName("play_place")
    private  String playPlace;
    @SerializedName("play_address")
    private  String playAddress;
    @SerializedName("play_picture")
    private  String playPicture;
    @SerializedName("play_price_adult")
    private  int priceAdult;
    @SerializedName("play_price_student")
    private  int priceStudent;
    @SerializedName("play_price_children")
    private  int priceChildren;
    @SerializedName("play_price_old_infirm")
    private  int priceOldInfirm;
    @SerializedName("play_price")
    private  String pricePlay;
    @SerializedName("play_call")
    private  String playCall;
    @SerializedName("play_site")
    private  String playSite;
    @SerializedName("play_detail_info")
    private  String playDetailInfo;
    @SerializedName("play_ticket_site1")
    private  String playTicketSite1;
    @SerializedName("play_ticket_site2")
    private  String playTicketSite2;

    private int playClicked;
    private int playLiked;
    private int playShared;
    private String playComments;

    public int getPlayId() {
        return playId;
    }

    public void setPlayId(int playId) {
        this.playId = playId;
    }

    public int getPlayCode() {return playCode; }

    public void setPlayCode(int playCode) {
        this.playCode = playCode;
    }

    public String getPlayTitle() {
        return playTitle;
    }

    public String getCloseDate() {
        return closeDate;
    }

    public void setCloseDate(String closeDate) {
        this.closeDate = closeDate;
    }

    public String getPlayTime() {
        return playTime;
    }

    public void setPlayTime(String playTime) {
        this.playTime = playTime;
    }

    public String getPlayRunTime() {
        return playRunTime;
    }

    public void setPlayRunTime(String playRunTime) {
        this.playRunTime = playRunTime;
    }

    public void setPlayTitle(String playTitle) {
        this.playTitle = playTitle;
    }

    public String getPlayActors() {
        return playActors;
    }

    public void setPlayActors(String playActors) {
        this.playActors = playActors;
    }

    public String getStartDate() {
        return startDate;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public String getEndDate() {
        return endDate;
    }

    public void setEndDate(String endDate) {
        this.endDate = endDate;
    }

    public String getPlayPlace() {
        return playPlace;
    }

    public void setPlayPlace(String playPlace) {
        this.playPlace = playPlace;
    }

    public String getPlayAddress() {
        return playAddress;
    }

    public void setPlayAddress(String playAddress) {
        this.playAddress = playAddress;
    }

    public String getPlayPicture() {
        return playPicture;
    }

    public void setPlayPicture(String playPicture) {
        this.playPicture = playPicture;
    }

    public int getPriceAdult() {
        return priceAdult;
    }

    public void setPriceAdult(int priceAdult) {
        this.priceAdult = priceAdult;
    }

    public int getPriceStudent() {
        return priceStudent;
    }

    public void setPriceStudent(int priceStudent) {
        this.priceStudent = priceStudent;
    }

    public int getPriceChildren() {
        return priceChildren;
    }

    public void setPriceChildren(int priceChildren) {
        this.priceChildren = priceChildren;
    }

    public int getPriceOldInfirm() {
        return priceOldInfirm;
    }

    public void setPriceOldInfirm(int priceOldInfirm) {
        this.priceOldInfirm = priceOldInfirm;
    }

    public String getPlayCall() {
        return playCall;
    }

    public void setPlayCall(String playCall) {
        this.playCall = playCall;
    }

    public String getPlaySite() {
        return playSite;
    }

    public void setPlaySite(String playSite) {
        this.playSite = playSite;
    }

    public int getPlayClicked() {
        return playClicked;
    }

    public void setPlayClicked(int playClicked) {
        this.playClicked = playClicked;
    }

    public int getPlayLiked() {
        return playLiked;
    }

    public void setPlayLiked(int playLiked) {
        this.playLiked = playLiked;
    }

    public int getPlayShared() {
        return playShared;
    }

    public void setPlayShared(int playShared) {
        this.playShared = playShared;
    }

    public String getPlayComments() {
        return playComments;
    }

    public void setPlayComments(String playComments) {
        this.playComments = playComments;
    }

    public String getPricePlay() {
        return pricePlay;
    }

    public void setPricePlay(String pricePlay) {
        this.pricePlay = pricePlay;
    }

    public String getPlayDetailInfo() {
        return playDetailInfo;
    }

    public void setPlayDetailInfo(String playDetailInfo) {
        this.playDetailInfo = playDetailInfo;
    }

    public String getPlayTicketSite1() {
        return playTicketSite1;
    }

    public void setPlayTicketSite1(String playTicketSite1) {
        this.playTicketSite1 = playTicketSite1;
    }

    public String getPlayTicketSite2() {
        return playTicketSite2;
    }

    public void setPlayTicketSite2(String playTicketSite2) {
        this.playTicketSite2 = playTicketSite2;
    }
}

File: BC_SERVER/application/models/PlayModel.php
<?php
class PlayModel extends CI_Model {

    function __construct(){
        parent::__construct();
        $this->load->database();
        $this->load->helper('date');
    }
    function changeDateTimeFormat($date){
        $date = strtotime($date);
        $date = date('y.m.d',$date);
        return $date;
    }
    function getPlayInfo($playId){
        $this->db->select('hb.play_id, hb.play_code, hb.play_title, hb.play_actors, hb.start_date, hb.end_date, hb.play_time, hb.play_run_time, hb.play_place, hb.play_address, hb.play_picture,hb.play_price_adult, hb.play_price_student, hb.play_price_children, hb.play_price_old_infirm,hb.play_price, hb.play_call, hb.play_site, hb.play_detail_info, hb.play_ticket_site1, hb.play_ticket_site2');
        $this->db->from('play as hb');
        $this->db->where('hb.play_id',$playId);
        $query = $this->db->get();
        $pos = 0;
        //$data = array();
        foreach($query->result() as $row){
            $data[$pos]['play_id'] = $row->play_id;
            $data[$pos]['play_code'] = $row->play_code;
            $data[$pos]['play_title'] = $row->play_title;
            $data[$pos]['play_actors'] = $row->play_actors;
            $data[$pos]['start_date'] = $this->changeDateTimeFormat($row->start_date);
            $data[$pos]['end_date'] = $this->changeDateTimeFormat($row->end_date);
            $data[$pos]['play_time'] = $row->play_time;
            $data[$pos]['play_run_time'] = $row->play_run_time;
            $data[$pos]['play_place'] = $row->play_place;
            $data[$pos]['play_address'] = $row->play_address;
            $data[$pos]['play_picture'] = $row->play_picture;
            $data[$pos]['play_price_adult'] = $row->play_price_adult;
            $data[$pos]['play_price_student'] = $row->play_price_student;
            $data[$pos]['play_price_children'] = $row->play_price_children;
            $data[$pos]['play_price_old_infirm'] = $row->play_price_old_infirm;
            $data[$pos]['play_price'] = $row->play_price;
            $data[$pos]['play_call'] = $row->play_call;
            $data[$pos]['play_site'] = $row->play_site;
            $data[$pos]['play_detail_info'] = $row->play_detail_info;
            $data[$pos]['play_ticket_site1'] = $row->play_ticket_site1;
            $data[$pos++]['play_ticket_site2'] = $row->play_ticket_site2;
        }
        return $data[0];
    }

    function getplayList($page){
        $this->db->select('hb.play_id, hb.play_code, hb.play_title, hb.play_actors, hb.start_date, hb.end_date, hb.play_time, hb.play_run_time, hb.play_place, hb.play_address, hb.play_picture,hb.play_price_adult, hb.play_price_student, hb.play_price_children, hb.play_price_old_infirm,hb.play_price, hb.play_call, hb.play_site, hb.play_detail_info, hb.play_ticket_site1, hb.play_ticket_site2');
        $this->db->from('play as hb');
        $this->db->order_by('hb.play_id','DESC');
        //$this->db->limit(2,$page*2);
        $query = $this->db->get();
        $pos = 0;
        foreach($query->result() as $row){
            $data[$pos]['play_id'] = $row->play_id;
            $data[$pos]['play_code'] = $row->play_code;
            $data[$pos]['play_title'] = $row->play_title;
            $data[$pos]['play_actors'] = $row->play_actors;
            $data[$pos]['start_date'] = $this->changeDateTimeFormat($row->start_date);
            $data[$pos]['end_date'] = $this->changeDateTimeFormat($row->end_date);
            $data[$pos]['play_time'] = $row->play_time;
            $data[$pos]['play_run_time'] = $row->play_run_time;
            $data[$pos]['play_place'] = $row->play_place;
            $data[$pos]['play_address'] = $row->play_address;
            $data[$pos]['play_picture'] = $row->play_picture;
            $data[$pos]['play_price_adult'] = $row->play_price_adult;
            $data[$pos]['play_price_student'] = $row->play_price_student;
            $data[$pos]['play_price_children'] = $row->play_price_children;
            $data[$pos]['play_price_old_infirm'] = $row->play_price_old_infirm;
            $data[$pos]['play_price'] = $row->play_price;
            $data[$pos]['play_call'] = $row->play_call;
            $data[$pos]['play_site'] = $row->play_site;
            $data[$pos]['play_detail_info'] = $row->play_detail_info;
            $data[$pos]['play_ticket_site1'] = $row->play_ticket_site1;
            $data[$pos++]['play_ticket_site2'] = $row->play_ticket_site2;
        }
        return $data;
    }
    
    function addClickNumToPlay($playId){
        $this->db->set('play_clicked', 'play_clicked+1', FALSE);
        $this->db->where('play_id', $playId);
        $result = $this->db->update('play'); 
        return $result;
    }

    function getGlobalFavoritePlay(){
        $this->db->select('hb.play_id, hb.play_code, hb.play_title, hb.play_actors, hb.start_date, hb.end_date, hb.play_time, hb.play_run_time, hb.play_place, hb.play_address, hb.play_picture,hb.play_price_adult, hb.play_price_student, hb.play_price_children, hb.play_price_old_infirm,hb.play_price, hb.play_call, hb.play_site, hb.play_detail_info, hb.play_ticket_site1, hb.play_ticket_site2');
        $this->db->from('play as hb');
        $this->db->order_by('hb.play_clicked', 'DESC');
        $query = $this->db->get();
        $pos = 0;
        foreach($query->result() as $row){
            $data[$pos]['play_id'] = $row->play_id;
            $data[$pos]['play_code'] = $row->play_code;
            $data[$pos]['play_title'] = $row->play_title;
            $data[$pos]['play_actors'] = $row->play_actors;
            $data[$pos]['start_date'] = $this->changeDateTimeFormat($row->start_date);
            $data[$pos]['end_date'] = $this->changeDateTimeFormat($row->end_date);
            $data[$pos]['play_time'] = $row->play_time;
            $data[$pos]['play_run_time'] = $row->play_run_time;
            $data[$pos]['play_place'] = $row->play_place;
            $data[$pos]['play_address'] = $row->play_address;
            $data[$pos]['play_picture'] = $row->play_picture;
            $data[$pos]['play_price_adult'] = $row->play_price_adult;
            $data[$pos]['play_price_student'] = $row->play_price_student;
            $data[$pos]['play_price_children'] = $row->play_price_children;
            $data[$pos]['play_price_old_infirm'] = $row->play_price_old_infirm;
            $data[$pos]['play_price'] = $row->play_price;
            $data[$pos]['play_call'] = $row->play_call;
            $data[$pos]['play_site'] = $row->play_site;
            $data[$pos]['play_detail_info'] = $row->play_detail_info;
            $data[$pos]['play_ticket_site1'] = $row->play_ticket_site1;
            $data[$pos++]['play_ticket_site2'] = $row->play_ticket_site2;
        }
        return $data;
    }
}
?>